src/acb_poly/compose_series.c

```diff
@@ -23,7 +23,24 @@ _acb_poly_compose_series(acb_ptr res, acb_srcptr poly1, slong len1,
     }
     else if (!_acb_vec_is_finite(poly1, len1) || !_acb_vec_is_finite(poly2, len2))
     {
-        _acb_vec_indeterminate(res, n);
+        /* find k such that the first k coefficients of both poly1 and
+         * poly2 are finite */
+        slong k = 0;
+
+        while (acb_is_finite(poly1 + k) && acb_is_finite(poly2 + k))
```

Comment:
I believe this can read out of bounds if one polynomial is shorter than the position of the first nonfinite coefficient in the other?

Reply:
I think this should be fixed now! I also updated the tests so that they catch this.